src/walkTree.ts

```diff
@@ -16,15 +19,15 @@ const walkTree = <T>(source: T, revision: T): T => {
   }
 
   // Return revision if its a non object type
-  if (typeof source !== "object" || typeof revision !== "object") {
+  if (!isObject(source) || !isObject(revision)) {
     return revision;
   }
 
   // Gather keys. For arrays, its just a list of indexes
   // TODO (asif) See if it'll be quicker walk arrays instead of getting its keys
   type keyType = keyof T;
   const sourceKeys = Object.keys(source) as keyType[];
-  const revisionKeys = Object.keys(revision) as keyType[];
+  // const revisionKeys = Object.keys(revision) as keyType[];
```

Comment:
unneeded?

Reply:
Yeah, but I used it in the next CL; so just ignore.